tools/c7n_azure/c7n_azure/resources/sqldatabase.py

```diff
@@ -30,3 +42,173 @@ class resource_type(ArmResourceManager.resource_type):
             manager_name='sqlserver',
             annotate_parent=True
         )
+
+
+class BackupRetentionPolicyFilter(Filter):
+
+    schema = type_schema(
+        'backup-retention-policy',
+        **{
+            'op': {'enum': list(scalar_ops.keys())}
+        }
+    )
+
+    @staticmethod
+    def get_sql_server_name(sql_server_id):
+        parsed = parse_resource_id(sql_server_id)
+        return parsed.get('name')
+
+    def __init__(self, operations_property, retention_limit, data, manager=None):
+        super(BackupRetentionPolicyFilter, self).__init__(data, manager)
+        self.operations_property = operations_property
+        self.retention_limit = retention_limit
+
+    def get_backup_retention_policy(self, i):
+        client = self.manager.get_client()
+        get_operation = getattr(client, self.operations_property).get
+
+        resource_group_name = i.get('resourceGroup')
+        server_name = BackupRetentionPolicyFilter.get_sql_server_name(
+            i.get(ChildResourceQuery.parent_key))
```

Comment:
Do we need `.get()` here instead of `[]`? What happens if there is no `parent_key` or `name`?

Reply:
I'll need to put some better handling of those cases in this section. 

In a Filter, should an exception be thrown if some required data is missing?

Reply:
I'd like to hear @stefangordon and @aluong opinions also.

I think if this is something we really need to properly apply filter, raising exception (or just a failfast) is the right behavior. CC should not make any assumptions if there is not enough data

Reply:
Yeah, I would assume raising an exception is the correct thing to do. But, if there is an established (alternate) pattern for this case, I can follow that.

Reply:
updated to raise exceptions, and included some test cases for that.